src/selfplay/tournament.cc

```diff
@@ -324,7 +585,12 @@ void SelfPlayTournament::Worker() {
     {
       Mutex::Lock lock(mutex_);
       if (abort_) break;
-      if (kTotalGames != -1 && games_count_ >= kTotalGames) break;
+      bool mirrored = player_options_[0].Get<bool>(kBookMirroredId.GetId());
+      if (kTotalGames != -1 && games_count_ >= kTotalGames ||
+          kTotalGames == -1 && !openings_.empty() &&
+              games_count_ >=
+                  static_cast<int>(openings_.size()) * (mirrored ? 2 : 1))
```

Comment:
Ah, so it's supposed to play openings once and exit? I thought it would just keep playing roung the list? And with random it doesn't make much sense either..

Reply:
This is overriding the meaning of -1 for convenience, I was originally considering introducing -2 to mean 'book length' and leaving -1 as infinite.
The user can always specific a stupidly huge number to get 'infinite' back though...

Reply:
I introduced the -2 idea.